client/engine/store/mockstore.go

```diff
@@ -136,13 +137,99 @@ func (ms MockStore) GetTwoPartyLedger(firstParty types.Address, secondParty type
 
 func (ms MockStore) GetObjectiveByChannelId(channelId types.Destination) (protocols.Objective, bool) {
 	// todo: locking
-	for _, obj := range ms.objectives {
+	for id, objJSON := range ms.objectives {
+		obj, err := decodeObjective(id, objJSON)
+
+		if err != nil {
+			return nil, false
+		}
+
 		for _, ch := range obj.Channels() {
 			if ch.Id == channelId {
+				obj, err = ms.populateChannelData(obj)
+
+				if err != nil {
+					return nil, false // todo: enrich w/ err return
+				}
+
 				return obj, true
 			}
 		}
 	}
 
 	return nil, false
 }
+
+// populateChannelData fetches stored Channel data relevent to the given
+// objective, attaches it to the objective, and returns the objective
+func (ms MockStore) populateChannelData(obj protocols.Objective) (protocols.Objective, error) {
+	id := obj.Id()
+
+	if dfo, isDirectFund := obj.(*directfund.Objective); isDirectFund {
+
+		ch, err := ms.getChannelById(dfo.C.Id)
+
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving channel data for objective %s: %w", id, err)
+		}
+
+		dfo.C = &ch
+
+		return dfo, nil
+
+	} else if vfo, isVirtualFund := obj.(*virtualfund.Objective); isVirtualFund {
+
+		v, err := ms.getChannelById(vfo.V.Id)
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving virtual channel data for objective %s: %w", id, err)
+		}
+		vfo.V = &channel.SingleHopVirtualChannel{Channel: v}
+
+		zeroAddress := types.Destination{}
+
+		if vfo.ToMyLeft != nil &&
+			vfo.ToMyLeft.Channel != nil &&
+			vfo.ToMyLeft.Channel.Id != zeroAddress {
+
+			left, err := ms.getChannelById(vfo.ToMyLeft.Channel.Id)
+			if err != nil {
+				return nil, fmt.Errorf("error retrieving left ledger channel data for objective %s: %w", id, err)
+			}
+			vfo.ToMyLeft.Channel = &channel.TwoPartyLedger{Channel: left}
+		}
+
+		if vfo.ToMyRight != nil &&
+			vfo.ToMyRight.Channel != nil &&
+			vfo.ToMyRight.Channel.Id != zeroAddress {
+			right, err := ms.getChannelById(vfo.ToMyRight.Channel.Id)
+			if err != nil {
+				return nil, fmt.Errorf("error retrieving right ledger channel data for objective %s: %w", id, err)
+			}
+			vfo.ToMyRight.Channel = &channel.TwoPartyLedger{Channel: right}
+
+		}
+
+		return vfo, nil
+
+	} else {
+		return nil, fmt.Errorf("objective %s did not correctly represent a known Objective type", id)
+	}
+}
+
+// decodeObjective is a helper which encapsulates the deserialization
```

Comment:
might be worth mentioning that `decodeObjective` doesn't handle populating channels (done by `populateChannelData`) in the godoc. Although it's a private function so it probably doesn't matter too much.